src/GitHub.TeamFoundation.14/Base/TeamExplorerNavigationItemBase.cs

```diff
@@ -35,10 +40,23 @@ public class TeamExplorerNavigationItemBase : TeamExplorerItemBase, ITeamExplore
             SubscribeToRepoChanges();
         }
 
-        public override async void Invalidate()
+        public override void Invalidate()
         {
             IsVisible = false;
-            IsVisible = await IsAGitHubRepo();
+            InvalidateAsync().Forget(log);
+        }
+
+        async Task InvalidateAsync()
+        {
+            var uri = ActiveRepoUri;
+            var isVisible = await IsAGitHubRepo(uri);
+            if (ActiveRepoUri != uri)
+            {
+                log.Information("Not setting button visibility because repository changed from {BeforeUrl} to {AfterUrl}", uri, ActiveRepoUri);
+                return;
```

Comment:
Can you explain what is happening to fall into this trap?

Reply:
I believe what is happening is that `Invalidate` is being called twice, once by Team Explorer and once by us when we detect that the repository has changed.

The problem happens when Team Explorer calls `Invalidate` before we know the repository has actually changed. We call an `async` method to check is the original (non-GitHub) repository is a GitHub/GHE repository.

Before this async method returns we're notified that the repository has changed and fire off another async method to check if the new repository is visible. This returns immediately because it's a `github.com` domain so we don't need to probe for Enterprise. The first call would then return having established that the original repository wasn't a GitHub repo, stomping over the new/correct visible state of the PRs icon.

I hope that makes sense. 😉

Reply:
It took me a few reads but i got it.